config.py

```
@@ -0,0 +1,418 @@
+            return {}
+        
+        try:
+            with open(self.config_file, 'r', encoding='utf-8') as f:
+                if self.config_file.endswith('.yaml') or self.config_file.endswith('.yml'):
+                    return yaml.safe_load(f) or {}
+                elif self.config_file.endswith('.json'):
+                    return json.load(f)
+                else:
+                    raise ConfigurationError(f"Unsupported config file format: {self.config_file}")
+        except Exception as e:
+            logger.warning(f"Failed to load config file {self.config_file}: {e}")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)